scripts/generate-changelog.js

```diff
@@ -0,0 +1,90 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const readline = require('readline')
+const replace = require('replace-in-file')
+const util = require('util')
+const exec = util.promisify(require('child_process').exec)
+const spawn = util.promisify(require('child_process').spawn)
+
+const name = require('emoji-name-map')
+
+const lernaConfig = require('../lerna.json')
+
+const CHANGELOG_FILE = 'CHANGELOG.md'
+
+async function main() {
+  const lastHashCmd = await exec('git rev-list --tags --max-count=1')
+  if (lastHashCmd.stderr) {
+    console.error(`error: ${stderr}`)
+    return
+  }
+
+  const oldTagCmd = await exec(`git describe --tags ${lastHashCmd.stdout}`)
+  if (oldTagCmd.stderr) {
+    console.error(`error: ${stderr}`)
+    return
+  }
+  const oldTag = oldTagCmd.stdout.replace(/\n$/, '')
+
+  const commitsCmd = await exec(`git log ${oldTag}..HEAD --pretty=format:"- %s"`)
+  if (commitsCmd.stderr) {
+    console.error(`error: ${stderr}`)
+    return
+  }
+
+  try {
+    await replace({
+      files: CHANGELOG_FILE,
+      from: new RegExp('# Changelog', 'g'),
+      to: `# Changelog\n\n## v${lernaConfig.version}\n\n${commitsCmd.stdout}`,
+    })
+  } catch (error) {
+    console.error('Error occurred:', error)
+  }
+
+  await emojiNameToUnicode()
```

Comment:
Could we not just parse emoji in `commitsCmd.stdout` ?

it would avoid to reopen and reparse the file every time